### ISSUE TITLE
Low validation accuracy if I split the training data manually when training with multi-machine

### ISSUE THREAD

**Issue by chenjx1005** [apache/incubator-mxnet#3325]
I training my model with a mpi cluster of 10 machines.  Because the training dataset is very large, distributing the whole `train.rec` to each machine requires much time and space. So I manually split the `train.lst` into 10 non-overlap text file such as `train_0.lst, train_1.lst....` . Then I use `im2rec.py` to make 10 ImageRecordIO dataset.

When training with multi machine, in each machine it will download `train_${OMPI_COMM_WORLD_RANK}.rec` and use it as the source of training iterator. I don't forget to change the parameters `num_parts` and  `part_index`  as `1, 0` of the training ImageRecordIter. 
I don't change the validation dataset. I distribute the complete `val.rec` to each machine.

So I believe the training procedure should be roughly same as when I use only one `train.rec`. 
But the result is disappointing. The training accuracy curve is good, even slightly better than previous. However the validation curve is much worse. 
![image](https://cloud.githubusercontent.com/assets/3807357/18622871/2e7c70e8-7e68-11e6-9d02-0ef70b3edfcf.png)
The blue curves are the training and val curves when I use one complete `train.rec` while red curves are those when using splited `train_0.rec, train_1.rec`....
Is there any mistake that I may have?

**Comment by tornadomeet** [comment]
@mli

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
